### PR TITLE
[v1.4] Graph Implementation

### LINKED ISSUE #124: [FEATURE] Graph visualisation for exercises
`graph n/NAME a/Y-AXIS sd/STARTDATE ed/ENDDATE`
Separate by exercise name

- logic by @ziyingli 
- gui by @Yonggiee

### LINKED ISSUE #235: [BUG] Calling graph on a non-existent exercise name throws uncaught exception
Steps to reproduce:

1. View any client with `view-c`
2. Call graph with a non-existent exercise name like `graph n/asd a/reps sd/20-02-2020 ed/20-03-2020`
3. Following success message is shown:
    Screenshot_20200408_100351
4. App throws uncaught `IndexOutOfBoundsException`

What is expected:

1. Success message to user should not be shown
2. Error should be thrown to user instead that says that the graph cannot be plotted because no exercises are found

### PR DESCRIPTION
**Pull request by ziyingli**
- [x] logic
- [x] tests
- [x] gui

- closes #124 
- closes #235

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/GraphCommand.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            + PREFIX_STARTDATE + "START_DATE "
+            + PREFIX_ENDDATE + "END_DATE \n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_EXERCISE_NAME + "push up "
+            + PREFIX_AXIS + "reps "
+            + PREFIX_STARTDATE + "01-01-2020 "
+            + PREFIX_ENDDATE + "01-04-2020";
+
+    public static final String MESSAGE_SUCCESS = "Graph displayed. Any new updates are not reflected. "
+        + "Currently displaying graph for exercise:\n%1$s";
```
Comment (AaronCQL) [inline src/main/java/seedu/address/logic/commands/GraphCommand.java]: This is very awkward phrasing. How about: "Displaying training graph for CLIENT's EXERCISE_NAME. Note that any new updates to the exercises (via add-e, edit-e, delete-e) will not be reflected on the current graph."

File: src/main/java/seedu/address/logic/commands/GraphCommand.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    public static final String MESSAGE_CLIENT_NOT_IN_VIEW = "You currently do not have a client in view, "
+        + "use the view-c command to view a client first";
+
+    public static final String MESSAGE_EXERCISE_NOT_IN_PERSONAL_BEST =
+        "This exercise does not have a personal best recorded";
+
+    public static final String MESSAGE_AXIS_UNSPECIFIED = "There was no axis specified.";
```
Comment (AaronCQL) [inline src/main/java/seedu/address/logic/commands/GraphCommand.java]: Remove these messages if they are not in use.

File: src/main/java/seedu/address/logic/commands/GraphCommand.java (hunk 1 above)
Comment (Yonggiee) [reply src/main/java/seedu/address/logic/commands/GraphCommand.java]: Remember the date too

Comment (ziyingli) [reply src/main/java/seedu/address/logic/commands/GraphCommand.java]: done

File: src/main/java/seedu/address/logic/commands/GraphCommand.java (hunk 2 above)
Comment (ziyingli) [reply src/main/java/seedu/address/logic/commands/GraphCommand.java]: updated

**Review by AaronCQL — APPROVED** [review]
LGTM

**Pull request merged**